pkg/tfgen/generate_csharp.go

```diff
@@ -1314,3 +1333,12 @@ func csharpDefaultValue(prop *variable) string {
 
 	return defaultValue
 }
+
+func isValueType(typ string) bool {
```

Comment:
nit: can we use a `kind` for the parameter type rather than `string`?

Reply:
I think I started with that, but then `isValueType` should also accept `wrapInput` to understand if it's nullable. It looked like a complication.